Doc/library/typing.rst

```diff
@@ -1628,6 +1619,40 @@ These are not used in annotations. They are building blocks for creating generic
 
    .. versionadded:: 3.8
 
+   .. versionchanged:: 3.12
+      The internal implementation of :func:`isinstance` checks against
+      runtime-checkable protocols now uses :func:`inspect.getattr_static`
+      to look up attributes (previously, :func:`hasattr` was used).
+      As a result, some objects which used to be considered instances
+      of a runtime-checkable protocol may no longer be considered instances
+      of that protocol on Python 3.12+, and vice versa.
+      Most users are unlikely to be affected by this change.
+
+   .. versionchanged:: 3.12
+      The members of a runtime-checkable protocol are now considered "frozen"
+      at runtime as soon as the class has been created. Monkey-patching
+      attributes onto a runtime-checkable protocol will still work, but will
+      have no impact on :func:`isinstance` checks comparing objects to the
+      protocol. For example:
+
+          >>> from typing import Protocol, runtime_checkable
+          >>> @runtime_checkable
+          ... class HasX(Protocol):
+          ...     x = 1
+          ...
+          >>> class Foo: ...
+          ...
+          >>> f = Foo()
+          >>> isinstance(f, HasX)
+          False
+          >>> f.x = 1
+          >>> isinstance(f, HasX)
+          True
+          >>> HasX.y = 2
+          >>> isinstance(f, HasX)  # unchanged, even though HasX now also has a "y" attribute
+          True
```

Comment:
I'm not convinced this example is worth the space it takes up on the page. IMO the description of the change is clear enough, and it's such an edge case we don't need to devote so much space to it. But I don't feel strongly.

Reply:
Yeah, I think you're right. I think what I'll do is keep the example in "What's new in Python 3.12", so that people specifically reading up on backwards-incompatible changes in Python 3.12 will see it, but I'll delete it from `typing.rst`. Sound good?

(The release manager can always delete it from "What's new" in a later stage if they don't think it's required in that document.)